### PR TITLE
lazily require gulp

### PR DESCRIPTION
**Pull request by Frizi**
enables usage only with externally provided gulp module

### CODE REVIEW

**Comment by OverZealous** [comment]
This doesn't pass the tests.

**Comment by Frizi** [comment]
Lame bug. Now tested this locally and should be ok.

File: index.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ function verifyTaskSets(gulp, taskSets, skipArrays) {
 }
 
 function runSequence(gulp) {
+	// enable gulp lazy loading
+	if(typeof gulp === 'function') {
```
Comment (OverZealous) [inline index.js]: I'm not sure why the extra level of indirection.  Why not simply do:
```js
if(gulp === undefined) {
    gulp = require('gulp');
}
```

Then the lower change can simply be `module.exports = runSequence;`

Comment (Frizi) [inline index.js]: Fair point, but i'll leave the bind to not change the api.

**Comment by Frizi** [comment]
Please review

File: index.js (hunk 1 above)
Comment (Frizi) [inline index.js]: Yes, this can also work, but you are changing public api much more.

**Comment by OverZealous** [comment]
Right, good point on the API.  I wasn't thinking too hard about it.

Now a bigger concern for me is that every time `runSequence()` is called, it `require`s gulp.  It's probably not a huge issue, though, given how runSequence is used.

I'll probably just accept this, but I need to think about it a little more.  It would be nice if you could squash the commits, too.

**Comment by Frizi** [comment]
Require is not a big issue, as it is internally cached very well. Each require call really just returns the same previously loaded instance. Only the first call is really a hassle, unless you play with this cache directly of course ;)

Will squash in a minute.

**Comment by Frizi** [comment]
Rebased

**Comment by OverZealous** [comment]
Published as 1.1.0

**Pull request merged**